pandas/sparse/array.py

```diff
@@ -152,9 +152,17 @@ def __new__(cls, data, sparse_index=None, index=None, kind='integer',
 
         # Create array, do *not* copy data by default
         if copy:
-            subarr = np.array(values, dtype=dtype, copy=True)
+            try:
+                # ToDo: Can remove this error handling when we actually
+                # support other dtypes
```

Comment:
hmm, maybe this should be more integrated with ``_sanitize_values`` (or that should be enhanced). not a big deal, but feel free to reorg code to make more general / readable code.